contribute/contribute.rst

```diff
@@ -266,149 +261,157 @@ A password will likely be requested and this is for the buildvm machine's root u
+   and so require an actual power cycle to return to their defaults.
+   The rockstor-hdparm.service is our way to re-establish custom config on boot-up.
+   The Rockstor Web-UI references this file itself for the current settings,
+   and no database component is thus linked to this configuration setting.
 
 .. _code_test:
 
-3. Code test
+2. Code test
 ~~~~~~~~~~~~
 
 At this point the Rockstor Web-UI should be available to verify your changes.
-In our example setup the URL from **laptop** would be :code:`https://buildvm/`.
+In our example setup, the URL from **laptop** would be :code:`https://buildvm/`.
 
 It is very important to ensure that your code changes survive a reboot.
-Sometimes, especially when db changes are made, this can be an issue.
+Sometimes, especially when database changes are made, this can be an issue.
 Be sure to check that the resulting build behaves as expected over:
 
-1. A config reset - removing /opt/rockstor-dev/.initrock and rebooting,
+1. A config reset - removing :code:`/opt/rockstor/.initrock` and rebooting
 2. Several reboots
 
-In (1.) above a db wipe is initiated helping to test the self-start code capability.
-See the initrock script and it's systemd trigger service: rockstor-init.
+In (1.) above, a database wipe is initiated helping to test the self-start code capability
+See the :code:`initrock.py` script and its systemd trigger service: :code:`rockstor-pre.service` for more details.
 
 We also have **automated tests** in place that cover our API's and core critical path functionality.
-It is expected that any changes to critical path code e.g. fs management / updates / Web-UI / services,
+It is expected that any changes to critical path code e.g. filesystem management/updates/Web-UI/services,
 include counterpart contributions to prove the expected function, if required.
-This is an oft neglected element in software development;
+This is an oft neglected element in software development
 but we are attempting to better our own standing in this regard.
 
 The following will run all tests following the source installation detailed above:
 
-.. code-block:: console
+.. code-block:: text
 
-    buildvm:~ # cd /opt/rockstor-dev/
-    buildvm:/opt/rockstor-dev # ./bin/test -v 2
+    buildvm:~ # cd /opt/rockstor/src/rockstor
+    buildvm:/opt/rockstor/src/rockstor # poetry run django-admin test -v 2
 
 All included tests, **numbering over 200**, are expected to pass;
-however it is always worth checking our `current issues <https://github.com/rockstor/rockstor-core/issues>`_
+however, it is always worth checking our `current issues <https://github.com/rockstor/rockstor-core/issues>`_
 for known failures in this area.
 
 A note on updating
 ``````````````````
 A source install will consider any rpm version to be an update.
-And this 'update' will necessarily delete all prior settings / db setup.
+And this 'update' will necessarily delete all prior settings/database setup.
```

Comment:
@FroggyFlox We actually have a differing behaviour here now. But it deserves its own issue so I've created one accordingly: #411 

It will need a little more thinking about/explanation so lets leave that change to the specific issue as everything else here is spot-on.

Reply:
Thanks a lot for #411. I was indeed unclear on that one so I took the safe route and didn't change much. Thank you for stepping up here; sorry for the added work, though.